src/uu/split/src/number.rs

```diff
@@ -467,17 +434,17 @@ mod tests {
         assert_eq!(format!("{}", num(9)), "09");
         assert_eq!(format!("{}", num(17)), "17");
         assert_eq!(format!("{}", num(10 * 9 - 1)), "89");
-        assert_eq!(format!("{}", num(10 * 9)), "9000");
-        assert_eq!(format!("{}", num(10 * 9 + 1)), "9001");
-        assert_eq!(format!("{}", num(10 * 99 - 1)), "9899");
-        assert_eq!(format!("{}", num(10 * 99)), "990000");
-        assert_eq!(format!("{}", num(10 * 99 + 1)), "990001");
+        assert_eq!(format!("{}", num(10 * 9)), "90");
+        assert_eq!(format!("{}", num(10 * 9 + 1)), "91");
+        assert_eq!(format!("{}", num(10 * 99 - 1)), "989");
+        assert_eq!(format!("{}", num(10 * 99)), "990");
+        assert_eq!(format!("{}", num(10 * 99 + 1)), "991");
```

Comment:
This was actually very intentional behaviour (and matched GNU), implemented in https://github.com/uutils/coreutils/pull/2859. Consider what happens to the filename sorting in your implementation:
```
some_file_98.txt
some_file_99.txt
some_file_100.txt
# gets sorted into this (due to lexical order of strings):
some_file_100.txt
some_file_98.txt
some_file_99.txt
```
The current algorithm, however, keeps `9`s at the start to ensure being sorted after earlier files.

Reply:
Okay so looking at this again I made an important discovery for this PR. With the `--numeric-suffix` option, the dynamic number is never used. It's always fixed width with a default width of 2. Here is an example:
```shell
$ split Cargo.lock Foo_ -l 1 --numeric-suffixes=80
split: output file suffixes exhausted
[ creates Foo_80 to Foo_99 ]
$ split Cargo.lock Foo_ -l 1 --numeric-suffixes=80 --suffix-length=4 
[ creates Foo_80 to Foo_3376 ]
$ split Cargo.lock Foo_ -l 1 -d          # warning creates a lot of files
[ creates Foo_00 up to Foo_992124 (which is 3307 files) ]
```

Reply:
OK - I'll take another pass at this later this week. It makes more sense now how it works, but it is a little crazy!